src/main/java/io/strimzi/test/container/StrimziKafkaCluster.java

```diff
@@ -147,10 +126,153 @@ public StrimziKafkaCluster(final int brokersNum, final ToxiproxyContainer proxyC
      * @param brokersNum number of brokers to be deployed
      * @param enableSharedNetwork enable Kafka cluster to use a shared Docker network.
      */
+    @Deprecated
     public StrimziKafkaCluster(final int brokersNum, final boolean enableSharedNetwork) {
         this(brokersNum, brokersNum, null, null, enableSharedNetwork);
     }
 
+    private StrimziKafkaCluster(StrimziKafkaClusterBuilder builder) {
+        this.brokersNum = builder.brokersNum;
+        this.enableSharedNetwork = builder.enableSharedNetwork;
+        this.network = this.enableSharedNetwork ? Network.SHARED : Network.newNetwork();
+        this.internalTopicReplicationFactor = builder.internalTopicReplicationFactor == 0 ? this.brokersNum : builder.internalTopicReplicationFactor;
+        this.additionalKafkaConfiguration = builder.additionalKafkaConfiguration;
+        this.proxyContainer = builder.proxyContainer;
+
+        validateBrokerNum(this.brokersNum);
+        validateInternalTopicReplicationFactor(this.internalTopicReplicationFactor);
+
+        this.zookeeper = new StrimziZookeeperContainer()
+            .withNetwork(this.network);
+
+        if (this.proxyContainer != null) {
+            this.proxyContainer.setNetwork(this.network);
+        }
+
+        startKafkaCluster(this.additionalKafkaConfiguration);
+    }
+
+    private void startKafkaCluster(final Map<String, String> additionalKafkaConfiguration) {
```

Comment:
Does this actually start it? doesn't the `start()` method do it? Should it be maybe called `prepareKafkaCluster` or something?

Reply:
Oh yea 🤦 ...`prepareKafkaCluster` sounds better. Thanks